src/docs/asciidoc/setting_up_clusters.adoc

```diff
@@ -583,32 +580,46 @@ include::{javasource}/ClientUCD.java[tag=clientucd]
 [[client-user-code-deployment-note]]
 ===== Important to Know
 
-Note that User Code Deployment should also be enabled on the
-members to use this feature.
+The members have to be configured in a specific way for the feature to work
+correctly:
+
+* User Code Deployment must be enabled on the members. Otherwise, the classes
+from the client will be ignored. Also blacklisted and non-whitelisted classes
+will be ignored.
+* All members must be providers, `provider-mode` must be set to
+`LOCAL_AND_CACHED_CLASSES` on all members.
+* No `provider-filter` must be configured.
+
+The client uploads the classes only to one member. If the members don't load
```

Comment:
@viliam-durina I have checked the code. There is a small difference than what I described to you earlier. 
Client uploads the classes to one member, and classes send the other members over this single member in the beginning.  
But those classes are not migrated to new joining members. The new joining members relies on the member to member user code deployment.

Reply:
my mind has parsing error here: :)
>and classes send the other members over this single member in the beginning

Is it that all current members somehow get the classes from the client when the client connects, only the newly joined members rely on member-to-member deployment? In that case a class-repository member will work...

Feel free to push to this PR with updated text. I'll then review and let you know if I understand it as a non-insider.